### PR TITLE
[Sidecar]LPS-106592-LPS-103577

### PR DESCRIPTION
**Pull request by tinatian**
Hi Shuyang,

I did not include the clean up of current embedded impl into this pull, only disabled EmbeddedElasticsearchConnection, search team will take care of the clean up once sidecar is in.

Please ignore the commit with prefix portal-search-elasticsearch7-impl: clustering, trying to figure out a better way to deal with cluster issues.

Known issues:
1.Dumping log4j.properties into temp folder
2.Random java.lang.InterruptedException on shutdown (https://github.com/shuyangzhou/liferay-portal/pull/8794)
3.Cluster rolling restart

Tina

### CODE REVIEW

**Comment by tinatian** [comment]
Test result see https://github.com/shuyangzhou/liferay-portal/pull/8852.

Tina

**Comment by slnn** [comment]
OK, got it

**Comment by slnn** [comment]
Hi @tinatian 

Encountered this issue when running startup performance test with this pull:
This issue only appeared with lpkg release bundle, reproduce steps:
1>Run ant all
2>Run  ant build-app-lpkg-release -Dliferay.home=${Bundle/root/path} under "${liferay-src}/modules"
3>Delete everything remaining in deploy, osgi/modules, osgi/portal, and osgi/static and check no manager folder under the webapps folder of tomcat
4>Set this properties in portal-ext.properties and startup bundle

> module.framework.properties.initial.system.check.enabled=false
> browser.launcher.url=


After digging into it found that:
In the lpkg release bundle of baseline(without the chanegs in this pull) , these *.lpkg files can be found in osgi/marketplace/

> Liferay CE Foundation - Liferay CE Connector to Elasticsearch 6 - API.lpkg
> Liferay CE Foundation - Liferay CE Connector to Elasticsearch 6 - Impl.lpkg
> Liferay CE Foundation - Liferay CE Connector to Elasticsearch 6 - SPI.lpkg

 

but with the lpkg release bundle of this pull, no related ES7 lpkg files be found in osgi/marketplace/, only a elasticsearch7 folder under the root directory of tomcat.

Part of the error:

`2020-02-16 05:41:40.374 ERROR [Framework Event Dispatcher: Equinox Container: 396d0110-ca76-4fd7-8699-e8959b78958b][Framework:93] FrameworkEvent ERROR 
org.osgi.framework.BundleException: Could not resolve module: com.liferay.batch.engine.service [78]_  Unresolved requirement: Import-Package: com.github.luben.zstd; resolution:="optional"_  Unresolved requirement: Import-Package: com.graphbuilder.curve; resolution:="optional"_  Unresolved requirement: Import-Package: com.graphbuilder.geom; resolution:="optional"_  Unresolved requirement: Import-Package: com.microsoft.schemas.office.powerpoint; resolution:="optional"_  Unresolved requirement: Import-Package: com.microsoft.schemas.office.word; resolution:="optional"_  Unresolved requirement: Import-Package: com.sun.javadoc; resolution:="optional"_  Unresolved requirement: Import-Package: com.sun.tools.javadoc; resolution:="optional"_  Unresolved requirement: Import-Package: net.sf.saxon; resolution:="optional"_  Unresolved requirement: Import-Package: net.sf.saxon.dom; resolution:="optional"_  Unresolved requirement: Import-Package: net.sf.saxon.om; resolution:="optional"_  Unresolved requirement: Import-Package: net.sf.saxon.query; resolution:="optional"_  Unresolved requirement: Import-Package: net.sf.saxon.sxpath; resolution:="optional"_  Unresolved requirement: Import-Package: net.sf.saxon.value; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.batik.anim.dom; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.batik.bridge; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.batik.ext.awt; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.batik.ext.awt.image.renderable; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.batik.gvt; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.batik.util; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.commons.math3.exception; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.commons.math3.linear; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.commons.math3.stat.descriptive.moment; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.commons.math3.stat.regression; resolution:="optional"_  Unresolved requirement: Import-Package: org.apache.jcp.xml.dsig.internal.dom; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.asn1; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.asn1.cmp; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.asn1.nist; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.asn1.ocsp; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.asn1.x500; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.asn1.x509; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.cert; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.cert.jcajce; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.cert.ocsp; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.cms; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.cms.bc; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.operator; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.operator.bc; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.tsp; resolution:="optional"_  Unresolved requirement: Import-Package: org.bouncycastle.util; resolution:="optional"_  Unresolved requirement: Import-Package: org.brotli.dec; resolution:="optional"_  Unresolved requirement: Import-Package: org.etsi.uri.x01903.v14; resolution:="optional"_  Unresolved requirement: Import-Package: org.openxmlformats.schemas.officeDocument.x2006.math; resolution:="optional"_  Unresolved requirement: Import-Package: org.openxmlformats.schemas.schemaLibrary.x2006.main; resolution:="optional"_  Unresolved requirement: Import-Package: org.tukaani.xz; resolution:="optional"_  Unresolved requirement: Import-Package: com.liferay.batch.engine; version="[3.0.0,4.0.0)"_    -> Export-Package: com.liferay.batch.engine; bundle-symbolic-name="com.liferay.batch.engine.api"; bundle-version="8.0.0"; version="3.0.0"; uses:="com.liferay.batch.engine.model,com.liferay.portal.kernel.model,com.liferay.portal.kernel.search,com.liferay.portal.kernel.search.filter,com.liferay.portal.odata.entity,com.liferay.portal.vulcan.accept.language,com.liferay.portal.vulcan.pagination"_       com.liferay.batch.engine.api [769]_         Unresolved requirement: Import-Package: com.liferay.portal.vulcan.pagination; version="[1.2.0,2.0.0)"_           -> Export-Package: com.liferay.portal.vulcan.pagination; bundle-symbolic-name="com.liferay.portal.vulcan.api"; bundle-version="7.2.0"; version="1.2.0"; uses:="com.fasterxml.jackson.annotation"_              com.liferay.portal.vulcan.api [868]_                Unresolved requirement: Import-Package: org.codehaus.stax2_         Unresolved requirement: Import-Package: com.liferay.portal.vulcan.accept.language; version="[1.1.0,2.0.0)"_           -> Export-Package: com.liferay.portal.vulcan.accept.language; bundle-symbolic-name="com.liferay.portal.vulcan.api"; bundle-version="7.2.0"; version="1.1.0"_  Unresolved requirement: Import-Package: com.liferay.batch.engine.configuration; version="[1.0.0,2.0.0)"_    -> Export-Package: com.liferay.batch.engine.configuration; bundle-symbolic-name="com.liferay.batch.engine.api"; bundle-version="8.0.0"; version="1.0.0"; uses:="aQute.bnd.annotation.metatype,com.liferay.portal.configuration.metatype.annotations"_ [Sanitized]
	at org.eclipse.osgi.container.Module.start(Module.java:444)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1682)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1662)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1624)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1555)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)`


More detailed log please see:
[startup.log](https://github.com/shuyangzhou/liferay-portal/files/4209421/startup.log)


Regards
Lily

**Comment by joshchong** [comment]
@slnn, it's intentional that the ES7 LPKGs are not showing up in osgi/marketplace since we're not planning on releasing ES7 as a bundled app just yet and will be releasing it on marketplace instead. So for your test, before deleting osgi/portal, you will have to package / deploy the ES7 jars as LPKGs into osgi/marketplace yourself.
Alternatively, this is just a guess, but you might be able to just change the [app.bnd for ES7](https://github.com/tinatian/liferay-portal/blob/LPS-106592-LPS-103577-40/modules/apps/portal-search-elasticsearch7/app.bnd) to match the [app.bnd for ES6](https://github.com/tinatian/liferay-portal/blob/LPS-106592-LPS-103577-40/modules/apps/portal-search-elasticsearch6/app.bnd) before running `build-app-lpkg-release` in your steps.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-api/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -52,6 +52,16 @@ remote-cluster-connection-id=Remote Cluster Connection ID
...
+sidecar-debug=Sidecar Debug
+sidecar-debug-help=Set to true to enable debug mode for sidecar process.
+sidecar-debug-settings=Sidecar Debug Settings
+sidecar-debug-settings-help=Set the jvm options used to debug sidecar process.
+sidecar-heartbeat-interval=Sidecar Heartbeat Interval
+sidecar-heartbeat-interval-help=Set heartbeat interval in milliseconds, which is used to detect the sidecar process healthiness.
+sidecar-home=Sidecar Home
+sidecar-home-help=Set the path of sidecar base folder, which is used to start sidecar process
+sidecar-jvm-options=Sidecar JVM Options
+sidecar-jvm-options-help=Set the jvm options used by sidecar processsidecar-jvm-options=Sidecar JVM Options
```
Comment (shuyangzhou) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-api/src/main/resources/content/Language.properties]: @tinatian you have a typo here.

**Comment by vicnate5** [comment]
@xbrianlee @joshchong Ran tests with ES7 sidecar disabled (ES default embedded). No unique issues found. https://github.com/vicnate5/liferay-portal/pull/1408

**Comment by joshchong** [comment]
Updated test fixes sent here:
https://github.com/tinatian/liferay-portal/pull/1716

**Pull request closed without merging**